mimic/model/maas_objects.py

```diff
@@ -0,0 +1,229 @@
+"""
+MaaS API data model
+"""
+
+import random
+
+
+def random_letters(len_min, len_max):
+    """
+    Makes a string with random letters (a-z).
+    """
+    return ''.join([chr(ord('a') + random.randint(0, 25))
+                    for _ in xrange(random.randint(len_min, len_max))])
+
+
+class Metric(object):
+    """
+    Models a MaaS metric type.
+    """
+    def __init__(self, name, type, clock, unit='other'):
+        """
+        Initializes the metric.
+        """
+        self.name = name
+        self.type = type
+        self._unit = unit
+        self._overrides = {}
+        self._clock = clock
+
+    def _override_key(self, **kwargs):
+        """
+        Computes a key used for hashing overrides, as a tuple of strings.
+        """
+        return (kwargs['entity_id'],
+                kwargs['check_id'],
+                kwargs.get('monitoring_zone', '__AGENT__'))
+
+    def set_override(self, **kwargs):
+        """
+        Sets the override metric for a given tenant, entity and check.
+
+        Override metrics are defined as a function which takes a timestamp
+        and returns the metric value.
+        """
+        self._overrides[self._override_key(**kwargs)] = kwargs['override_fn']
+
+    def clear_overrides(self):
+        """
+        Clears the override metric values.
+        """
+        self._overrides = {}
+
+    def _get_default_data(self):
```

Comment:
I don't think I quite get this; it seems like a `Metric` should have a single fixed value, or at least a fixed value for a particular point in time.  It looks here like you'll request the same metric and get a randomly different value?

The use of the `random` module all over the place worries me a little, too.  There are two use-cases for randomness, generally.  In one case, you want deterministic random values, in which case you generally want an instance of `random.Random` that has a given seed; in another case, you want truly unpredictable cryptographic randomness, in which case you want an instance of `random.SystemRandom`.  There isn't really a case where you want to use the global functions in the `random` module (and in fact there is a discussion going on right now in the python community about possibly deprecating those methods entirely).

I think that we may want `MimicCore` to have a source of randomness, a `random` attribute, like `clock`, which gets passed around to do these various randomness-generation things, so that users can select which source of randomness is appropriate for their test setup.  By default it should probably be `SystemRandom` so it's truly unpredictable, since most places we are using randomness are things that tests should be ignoring and able to cope with being non-deterministic…